plugins/vli/fu-vli-pd-device.c

```diff
@@ -419,19 +420,66 @@ fu_vli_pd_device_write_firmware (FuDevice *device,
 	if (!fu_vli_pd_device_write_reg (self, 0x0003, tmp | 0x44, error))
 		return FALSE;
 
-	/* erase */
-	fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_ERASE);
-	if (!fu_vli_device_spi_erase_all (FU_VLI_DEVICE (self), error))
-		return FALSE;
-
-	/* write in chunks */
-	fu_device_set_status (FU_DEVICE (self), FWUPD_STATUS_DEVICE_WRITE);
-	buf = g_bytes_get_data (fw, &bufsz);
-	if (!fu_vli_device_spi_write (FU_VLI_DEVICE (self),
-				      fu_vli_device_get_offset (FU_VLI_DEVICE (self)),
-				      buf, bufsz, error))
-		return FALSE;
+	/* Dual image on VL103 */
+	if (fu_vli_device_get_kind (FU_VLI_DEVICE (device)) == FU_VLI_DEVICE_KIND_VL103 &&
+		fu_device_has_custom_flag (device, "dual-image")) {
+		gsize sbufsz = 0;
+		const guint8 *sbuf = NULL;
+		guint16 crc_actual;
+		guint16 crc_file = 0x0;
+		g_autoptr(GBytes) spi_fw = NULL; 
```

Comment:
I think all of this indented section should probably be moved into a new static function, e.g. `fu_vli_pd_device_write_dual_firmware` or something.

Reply:
Where in the parade would you like me to put this new function? Directly above?

Reply:
Yes, that's perfect, thanks.